src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] arrayList;
+    private int size;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == arrayList.length) {
+            resize(size);
+        }
+        arrayList[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index
+                    + " out of ArrayListSize");
+        }
+
+        if (size == arrayList.length) {
+            resize(arrayList.length * 2);
+        }
+
+        for (int i = size; i > index; i--) {
+            arrayList[i] = arrayList[i - 1];
+        }
 
+        arrayList[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        int arrayPlusListSize = size + list.size();
+        if (arrayPlusListSize > arrayList.length) {
+            resize(arrayPlusListSize);
+        }
```

Comment:
redundant 
addAll is functionally equivalent to adding in a loop
resizing logic is in add already